relayer/src/foreign_client.rs

```diff
@@ -542,7 +545,7 @@ impl ForeignClient {
 
         assert!(!events.is_empty());
```

Comment:
Do we still need this assertion here? I guess we still expect the `events` vec to be non-empty but since we can now gracefully handle the case where it's not, perhaps we could convert it to a `debug_assert!`?

Reply:
I think it is fine to remove it. I checked the callers and none of them are really doing anything with the result returned. Otherwise it would depends on whether the other functions check on the events returned. If we want to do checking, it should still return an `Err` result instead of using `assert!` or `debug_assert!`.